pkg/util/encryption/multi.go

```diff
@@ -52,41 +53,58 @@ func NewMulti(ctx context.Context, serviceKeyvault azsecrets.Client, secretName,
 			}
 			for _, properties := range page.Value {
 				if properties != nil && properties.ID != nil && properties.Attributes != nil {
-					raw, err := serviceKeyvault.GetSecret(ctx, (*properties.ID).Name(), (*properties.ID).Version(), nil)
+					secretVersion := (*properties.ID).Version()
+					raw, err := serviceKeyvault.GetSecret(ctx, (*properties.ID).Name(), secretVersion, nil)
 					if err != nil {
 						return nil, err
 					}
 					version, err := azsecrets.ExtractBase64Value(raw)
 					if err != nil {
 						return nil, err
 					}
-					keys = append(keys, version)
+
+					aead, err = x.aeadFactory(ctx, version, secretVersion)
+					if err != nil {
+						return nil, err
+					}
+
+					m.openers = append(m.openers, aead)
 				}
 			}
 		}
+	}
 
-		for _, key := range keys {
-			aead, err = x.aeadFactory(ctx, key)
-			if err != nil {
-				return nil, err
-			}
+	return m, nil
+}
 
-			m.openers = append(m.openers, aead)
-		}
+func (c *multi) Name() string {
+	var openerNames []string
+
+	for _, i := range c.openers {
+		openerNames = append(openerNames, i.Name())
 	}
 
-	return m, nil
+	return fmt.Sprintf("Multi(sealer=%s, openers=%s)", c.sealer.Name(), strings.Join(openerNames, ","))
 }
 
-func (c *multi) Open(input []byte) (b []byte, err error) {
+func (c *multi) Open(input []byte) ([]byte, error) {
+	var errs []error
+
 	for _, opener := range c.openers {
-		b, err = opener.Open(input)
+		b, err := opener.Open(input)
 		if err == nil {
-			return
+			return b, nil
+		} else {
+			errs = append(errs, fmt.Errorf("%s: %w", opener.Name(), err))
 		}
 	}
 
-	return nil, err
+	errStrings := ""
+	for _, err := range errs {
+		errStrings = errStrings + fmt.Sprintf("\n\t* %s", err)
+	}
+
+	return nil, fmt.Errorf("no openers succeeded:%s", errStrings)
```

Comment:
Prefer if we use `var ErrorNoOpeners = errors.New("no openers succeeded")` and make use of `errors.Join` to wrap the errors. Just allows for better testing + reuse.